Tests/TuistDependenciesTests/SwiftPackageManager/Utils/SwiftPackageManagerControllerTests.swift

```diff
@@ -0,0 +1,347 @@
+import TSCBasic
+import TuistCore
+import TuistGraph
+import TuistSupport
+import XCTest
+
+@testable import TuistDependencies
+@testable import TuistSupportTesting
+
+final class SwiftPackageManagerControllerTests: TuistUnitTestCase {
+    private var subject: SwiftPackageManagerController!
+
+    override func setUp() {
+        super.setUp()
+
+        subject = SwiftPackageManagerController()
+    }
+
+    override func tearDown() {
+        subject = nil
+
+        super.tearDown()
+    }
+
+    func test_resolve() throws {
+        // Given
+        let path = try temporaryPath()
+        system.succeedCommand([
+            "swift",
+            "package",
+            "--package-path",
+            path.pathString,
+            "resolve",
+        ])
+
+        // When / Then
+        XCTAssertNoThrow(try subject.resolve(at: path))
+    }
+
+    func test_update() throws {
+        // Given
+        let path = try temporaryPath()
+        system.succeedCommand([
+            "swift",
+            "package",
+            "--package-path",
+            path.pathString,
+            "update",
+        ])
+
+        // When / Then
+        XCTAssertNoThrow(try subject.update(at: path))
+    }
+
+    func test_setToolsVersion_specificVersion() throws {
+        // Given
+        let path = try temporaryPath()
+        let version = "5.4"
+        system.succeedCommand([
+            "swift",
+            "package",
+            "--package-path",
+            path.pathString,
+            "tools-version",
+            "--set",
+            version,
+        ])
+
+        // When / Then
+        XCTAssertNoThrow(try subject.setToolsVersion(at: path, to: version))
+    }
+
+    func test_setToolsVersion_currentVersion() throws {
+        // Given
+        let path = try temporaryPath()
+        system.succeedCommand([
+            "swift",
+            "package",
+            "--package-path",
+            path.pathString,
+            "tools-version",
+            "--set-current",
+        ])
+
+        // When / Then
+        XCTAssertNoThrow(try subject.setToolsVersion(at: path, to: nil))
+    }
+
+    func test_loadPackageInfo() throws {
+        // Given
+        let path = try temporaryPath()
+        system.succeedCommand(
+            [
+                "swift",
+                "package",
+                "--package-path",
+                path.pathString,
+                "dump-package",
+            ],
+            output: Self.testPackageInfoJSON
+        )
+
+        // When
+        let packageInfo = try subject.loadPackageInfo(at: path)
+
+        // Then
+        XCTAssertEqual(packageInfo, Self.testPackageInfo)
+    }
+}
+
+extension SwiftPackageManagerControllerTests {
+    static let testPackageInfoJSON = """
```

Comment:
Could you move it to `TuistDependenicesTesting` framework. 

[Carthage's](https://github.com/tuist/tuist/blob/bb5534bc4def00e7ff28f8e8d1e294da158c07b6/Sources/TuistDependenciesTesting/Carthage/Models/CarthageVersionFile%2BTestData.swift) testing data.